modules/services/emacs.nix

```diff
@@ -44,6 +44,15 @@ in {
   options.services.emacs = {
     enable = mkEnableOption "the Emacs daemon";
 
+    shell = mkOption {
+      type = types.shellPackage;
+      default = pkgs.bash;
```

Comment:
Couldn't this use user's shell variable?

Reply:
It's possible.

I just don't want to deal the weird quoting syntax with both shell and systemd. Writing a start script like what the nixos option `systemd.services.<name>.script` does is probably easier and clearer.

Feel free to open another PR. I would love to see either of them merged.